Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveyModule.kt

```diff
@@ -1,9 +1,62 @@
 package de.rki.coronawarnapp.datadonation.survey
 
+import android.content.Context
 import dagger.Module
+import dagger.Provides
+import dagger.Reusable
 import de.rki.coronawarnapp.datadonation.survey.consent.SurveyConsentModule
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationApiV1
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationHttpClient
+import de.rki.coronawarnapp.environment.datadonation.DataDonationCDNServerUrl
+import de.rki.coronawarnapp.http.HttpClientDefault
+import de.rki.coronawarnapp.http.RestrictedConnectionSpecs
+import de.rki.coronawarnapp.util.di.AppContext
+import okhttp3.Cache
+import okhttp3.ConnectionSpec
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.protobuf.ProtoConverterFactory
+import java.io.File
+import javax.inject.Singleton
 
 @Module(
     includes = [SurveyConsentModule::class]
 )
-class SurveyModule
+class SurveyModule {
+
+    @Reusable
+    @DataDonationHttpClient
```

Comment:
There is already a more specific `OkHttpClient` with the annotation `@DataDonationCDNHttpClient` which should be used here.

-> `@HttpClientDefault` -> `@DataDonationCDNHttpClient`.

If you need a specific HTTP client for the surveys, does it need to be exposed via custom label to the whole app, or is it an implementation detail of `provideDataDonationApi` and the only the that needs to be exposed via the injection graph is `DataDonationApiV1`?

Reply:
reusing it!